test/opt/test_unify_const.cpp

```diff
@@ -0,0 +1,880 @@
+          {
+            "%float_var = OpVariable %_pf_float Function",
+            "OpStore %float_var %float_nan_1_duplicate",
+            "OpStore %float_var %float_nan_2_duplicate",
+          },
+          // remapped names
+          {
+            "OpName %float_nan_1 \"float_nan_1_duplicate\"",
+            "OpName %float_nan_2 \"float_nan_2_duplicate\"",
+          },
+        },
+        // null values
+        {
+          // preserved constants
+          {
+            "%bool_null = OpConstantNull %bool",
+            "%signed_null = OpConstantNull %int",
+            "%signed_64_null = OpConstantNull %int64",
+            "%float_null = OpConstantNull %float",
+            "%double_null = OpConstantNull %double",
+            // zero-valued constants will not be unified with the equivalent
+            // null constants.
+            "%signed_zero = OpConstant %int 0",
+          },
+          // use preserved constants in main
+          {
+            "%bool_var = OpVariable %_pf_bool Function",
+            "OpStore %bool_var %bool_null",
+            "%int_var = OpVariable %_pf_int Function",
+            "OpStore %int_var %signed_null",
+            "%int64_var = OpVariable %_pf_int64 Function",
+            "OpStore %int64_var %signed_64_null",
+            "%float_var = OpVariable %_pf_float Function",
+            "OpStore %float_var %float_null",
+            "%double_var = OpVariable %_pf_double Function",
+            "OpStore %double_var %double_null",
+          },
+          // duplicated constants
+          {
+            "%bool_null_duplicate = OpConstantNull %bool",
+            "%signed_null_duplicate = OpConstantNull %int",
+            "%signed_64_null_duplicate = OpConstantNull %int64",
+            "%float_null_duplicate = OpConstantNull %float",
+            "%double_null_duplicate = OpConstantNull %double",
+          },
+          // use duplicated constants in main
+          {
+            "%bool_var = OpVariable %_pf_bool Function",
+            "OpStore %bool_var %bool_null_duplicate",
+            "%int_var = OpVariable %_pf_int Function",
+            "OpStore %int_var %signed_null_duplicate",
+            "%int64_var = OpVariable %_pf_int64 Function",
+            "OpStore %int64_var %signed_64_null_duplicate",
+            "%float_var = OpVariable %_pf_float Function",
+            "OpStore %float_var %float_null_duplicate",
+            "%double_var = OpVariable %_pf_double Function",
+            "OpStore %double_var %double_null_duplicate",
+          },
+          // remapped names
+          {
+            "OpName %bool_null \"bool_null_duplicate\"",
+            "OpName %signed_null \"signed_null_duplicate\"",
+            "OpName %signed_64_null \"signed_64_null_duplicate\"",
+            "OpName %float_null \"float_null_duplicate\"",
+            "OpName %double_null \"double_null_duplicate\"",
+          },
+        },
+        // TODO(qining): support constant sampler
```

Comment:
a test of Constant Sampler is added. Lucky we don't have Null Constant of this type.